pkg/cache/cache.go

```diff
@@ -140,10 +143,44 @@ func (c *serviceAccountCache) ToJSON() string {
 	return string(contents)
 }
 
+func getAccountId() (string, error) {
+	sess, err := session.NewSession()
+	if err != nil {
+		klog.Errorf("Could not create new session", err.Error())
+		return "", err
+	}
+
+	metadata := ec2metadata.New(sess)
+	identity, err := metadata.GetInstanceIdentityDocument()
+	if err != nil {
+		klog.Errorf("Could not get instance identity document", err.Error())
+		return "", err
+	}
+
+	accountID := identity.AccountID
+
+	return accountID, nil
+}
+
 func (c *serviceAccountCache) addSA(sa *v1.ServiceAccount) {
 	arn, ok := sa.Annotations[c.annotationPrefix+"/"+pkg.RoleARNAnnotation]
+
 	resp := &CacheResponse{}
 	if ok {
+		parsedARN, err := awsarn.Parse(arn)
```

Comment:
I think this block needs to be rewritten.

awsarn.Parse(arn) returns an error if it can't find the prefix `arn:` ...
we need to check that error via awserror

If this error matches we will call getAccountId and add the arn. Any other error we need to fail here